docs/ds/sbt.md

```diff
@@ -1 +1,361 @@
+  //     |                     |                      |
+  //     N                     N                     [M]
+  //    / \    r-rotate(R)    / \     l-rotate(N)    / \
+  //  <L>  R   ==========>  <L> [M]   ==========>   N   R
+  //      /                       \                /
+  //    [M]                        R             <L>
+  // clang-format on
+  rotateRight(node->right);
+  rotateLeft(node);
+  fixBalance(node->left);
+  fixBalance(node->right);
+  fixBalance(node);
+  return;
+}
+```
+
+#### Case 2: $size(N.left) \lt size(N.right.right)$
+
+```c++
+if (size(node->right->right) > size(node->left)) {
+  // clang-format off
+  //     |                       |
+  //     N                       R
+  //    / \     l-rotate(N)     / \
+  //  <L>  R    ==========>    N  [M]
+  //        \                 /
+  //        [M]             <L>
+  // clang-format on
+  rotateLeft(node);
+  fixBalance(node->left);
+  fixBalance(node);
+  return;
+}
+```
+
+#### Case 3: $size(N.right) \lt size(N.left.left)$
+
+```c++
+if (size(node->left->left) > size(node->right)) {
+  // clang-format off
+  //       |                       |
+  //       N                       L
+  //      / \     r-rotate(N)     / \
+  //     L  <R>   ==========>   [M]  N
+  //    /                             \
+  //  [M]                             <R>
+  // clang-format on
+  rotateRight(node);
+  fixBalance(node->right);
+  fixBalance(node);
+  return;
+}
+```
+
+#### Case 4: $size(N.right) \lt size(N.left.right)$
+
+```c++
+if (size(node->left->right) > size(node->right)) {
+  // clang-format off
+  //     |                     |                      |
+  //     N                     N                     [M]
+  //    / \    l-rotate(L)    / \     r-rotate(N)    / \
+  //   L  <R>  ==========>  [M] <R>   ==========>   L   N
+  //    \                   /                            \
+  //    [M]                L                             <R>
+  // clang-format on
+  rotateLeft(node->left);
+  rotateRight(node);
+  fixBalance(node->left);
+  fixBalance(node->right);
+  fixBalance(node);
+  return;
+}
+```
+
+## 操作
+
+### 插入
+
+SBT 的插入操作需要在完成普通二叉搜索树的插入操作的基础上递归地进行节点`size`字段的更新及平衡维护. 示例代码如下:
+
+```c++
+if (compare(key, node->key)) {
+  /* key < node->key */
+  if (node->left == nullptr) {
+    node->left = Node::from(key, value);
+    node->updateSize();
+  } else {
+    insert(node->left, key, value, replace);
+    node->updateSize();
+    fixBalance(node);
+  }
+} else {
+  /* key > node->key */
+  if (node->right == nullptr) {
+    node->right = Node::from(key, value);
+    node->updateSize();
+  } else {
+    insert(node->right, key, value, replace);
+    node->updateSize();
+    fixBalance(node);
+  }
+}
+```
+
+### 删除
+
+根据 Size Balanced Tree 的提出者陈启峰在其论文中对于删除操作的描述:
+
+> It can result in a destroyed SBT. But with the insertion above, a BST is still kept at the height of $O(logn)$ where $n$ is the total number of insertions, not the current size.
```

Comment:
```suggestion
> It can result in a destroyed SBT. But with the insertion above, a BST is still kept at the height of $O(\log n)$ where $n$ is the total number of insertions, not the current size.
```

Reply:
all fixed